File: library/src/main/java/com/lazygeniouz/house/ads/listener/NativeAdListener.java
/*
 * Created by Darshan Pandya.
 * @itznotabug
 * Copyright (c) 2018.
 */

package com.lazygeniouz.house.ads.listener;

import android.view.View;

@SuppressWarnings("unused")
public interface NativeAdListener {

    void onAdLoaded();
    void onAdLoadFailed(Exception e);

    interface CallToActionListener{
        void onCallToActionClicked(View view);
    }
}

File: library/src/main/java/com/lazygeniouz/house/ads/listener/NativeAdListener.kt
/*
 * Created by Darshan Pandya. (@itznotabug)
 * Copyright (c) 2018-2020.
 */

package com.lazygeniouz.house.ads.listener

import android.view.View

interface NativeAdListener {
    fun onAdLoaded()
    fun onAdFailedToLoad(exception: Exception)
}

interface NativeActionListener {
    fun onClick(view: View)
}
